modules/setting/mirror.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package setting
+
+import (
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+)
+
+var (
+	// Mirror settings
+	Mirror = struct {
+		Enabled         bool
+		DisableNewPull  bool
+		DisableNewPush  bool
+		DefaultInterval time.Duration
+		MinInterval     time.Duration
+	}{
+		Enabled:         true,
+		DisableNewPull:  false,
+		DisableNewPush:  false,
+		MinInterval:     10 * time.Minute,
+		DefaultInterval: 8 * time.Hour,
+	}
+)
+
+func newMirror() {
+	// Handle old configuration through `[repository]` `DISABLE_MIRRORS`
+	// - please note this was badly named and only disabled the creation of new pull mirrors
+	if Cfg.Section("repository").Key("DISABLE_MIRRORS").MustBool(false) {
+		log.Warn("Deprecated DISABLE_MIRRORS config is used, please change your config and use the options within the [mirror] section")
+		// TODO: enable on v1.17.0: log.Error("Deprecated fallback used, will be removed in v1.18.0")
+		Mirror.DisableNewPull = true
+	}
+	if err := Cfg.Section("mirror").MapTo(&Mirror); err != nil {
+		log.Fatal("Failed to map Mirror settings: %v", err)
+	}
+
+	if !Mirror.Enabled {
+		Mirror.DisableNewPull = true
+		Mirror.DisableNewPush = true
+	}
+
+	if Mirror.MinInterval.Minutes() < 1 {
+		log.Warn("Mirror.MinInterval is too low, set to 1 minute")
+		Mirror.MinInterval = 1 * time.Minute
+	}
+	if Mirror.DefaultInterval < Mirror.MinInterval {
+		log.Warn("Mirror.DefaultInterval is less than Mirror.MinInterval, set to 8 hours")
+		Mirror.DefaultInterval = time.Hour * 8
```

Comment:
What happens when the min interval is larger than 8 hours?
Then there will be an interesting bug where the default interval is smaller than the min interval.
Could be fixed by using `Math.max(time.Hour * 8, Mirror.MinInterval)` or whatever the exact syntax in go is.

Reply:
Also, the warning one statement above should then be renamed to `… set to at least 8 hours`

Reply:
did had similar thoughts, but let it be as is and just moved that code ... - will add a check

Reply:
done - but if you really have this case - I would argue the admin did this on propose ...